src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +16,33 @@
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
+            transaction.commit();
+            return country;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot ADD country to Database", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        try(Session session = factory.openSession()) {
```

Comment:
Similar to the `add` method, ensure that `sessionFactory` is used instead of `factory` to open sessions. This will maintain consistency and avoid potential issues.